docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+`WerkIt#startContinuousUserPrompt()` calls `Parser#parseUserInput()` to parse the user's input and create a
+an object that is a subclass of the `Command` class. If there is no issue with the formatting of the user's input,
+this subclass-of-`Command` object is returned to `WerkIt#startContinuousUserPrompt()`.
+
+> A detailed implementation of the parsing and creation of subclass-of-`Command` object process can be found in
+'[Parsing User Input and Getting the Right Command](#parsing-user-input-and-getting-the-right-command)'.
+
+Next, `WerkIt#startContinuousUserPrompt()` calls on the `execute()` method of the subclass-of-`Command` object to
+perform the user's requested action. If the execution goes smoothly, this completes the user's inputted command.
+This process is repeated until the user enters `exit`, which will terminate the loop, call `UI#printGoodbye()` to
+print a goodbye message to the user, before handing control back to `Main#main` to end the program.
+
+#### Design Considerations
+* `WerkIt#startContinuousUserPompt()` has a boolean flag `isFirstPrompt`. This flag allows WerkIt to
+print a different prompt each time the application starts up, before defaulting to a different prompt message
+for subsequent prompts.
+   * When the user starts the application, `isFirstPrompt` is set to `true` and thus, the prompt will be
+  ```
+  ----------------------------------------------------------------------
+  Now then, what can I do for you today?
+  (Need help? Type 'help' for a guide!)
+  ----------------------------------------------------------------------
+  >
+  ```
+  * Subsequent prompts in that app's session will be
+  ```
+  ----------------------------------------------------------------------
+  What's next?
+  ----------------------------------------------------------------------
+  >
+  ```
+
+---
+
+### Parsing User Input and Getting the Right Command
+**_TODO_**: Explain how the app parses user input and determines which `Command` subclass object to instantiate.
+
+---
+
+### Exercise
+#### List Exercise
+
+
+If the user's command type is to list the exercises available, i.e. `exercise /list`, the
+`Parser#parseUserInput(String userInput)` method will parse the 'exercise' base word and proceed to create exercise related
+command using `Parser#createExerciseCommand(String userInput)` method. This method will further evaluate the
+exercise action, in this case, `/list` and call the constructor of `ExerciseCommand` class by passing relevant parameters related to the
+ExerciseCommand constructor. If the exercise action is null or incorrect, an InvalidCommandException will be thrown. Once the exercise command is created,
+this exercise command is executed via the `ExerciseCommand#execute()` method. As it is executed, the method will check the
+type of action to be executed, in this case, list. It will then list the exercises available for selection from the exerciseList using the `ExerciseList#printExerciseList()`.
```

Comment:
I agree with the other comment on this, perhaps it would be better to follow a more sequential approach to this, as your team has done in Workout. Visually, I think it may be more helpful to have the image before the text too, since most of it are describing the image.